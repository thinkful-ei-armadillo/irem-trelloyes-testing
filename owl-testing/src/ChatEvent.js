import React from 'react';

function ChatEvent(props){
    return (
        <div className="ChatEvent">
            <p>{props.type}</p>
            <p>{props.message}</p>
            <p>{props.timestamp}</p>
        </div>
    );
}

export default ChatEvent;