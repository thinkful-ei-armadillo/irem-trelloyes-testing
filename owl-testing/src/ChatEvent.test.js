import React from 'react';
import ReactDOM from 'react-dom';
import ChatEvent from './ChatEvent';
import renderer from 'react-test-renderer';

describe('<App />', () => {
it('renders without crashing', () => {
  const div = document.createElement('div');

  ReactDOM.render(<ChatEvent  
      type ="message"
      message ="Testing Chat Log"
      timestamp="1548852484247"
  />, div);

  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
    const tree = renderer
      .create(<ChatEvent 
      type ="message"
      message ="Testing Chat Log"
      timestamp="1548852484247" />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    })
})