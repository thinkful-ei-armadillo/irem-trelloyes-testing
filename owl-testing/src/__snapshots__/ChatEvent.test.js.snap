// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders the UI as expected 1`] = `
<div
  className="ChatEvent"
>
  <p>
    message
  </p>
  <p>
    Testing Chat Log
  </p>
  <p>
    1548852484247
  </p>
</div>
`;
